### PR TITLE
Added homi flag for kip71 compatible block

### PR DESCRIPTION
**Pull request by kjeom**
## Proposed changes

- Added kip71CompatibleBlock flag for homi.
- I tested the function by executing the homi and checked that it works well.
- After KIP-71 hardfork, verifyHeader() needs parent instance for checking the baseFee.
  - If the current block is genesis, the parent is also genesis.

## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by kjhman21 — APPROVED** [review]

File: consensus/istanbul/backend/engine.go [hunk 1]
```diff
@@ -172,7 +172,14 @@ func (sb *backend) computeSignatureAddrs(header *types.Header) error {
 // given engine. Verifying the seal may be done optionally here, or explicitly
 // via the VerifySeal method.
 func (sb *backend) VerifyHeader(chain consensus.ChainReader, header *types.Header, seal bool) error {
-	return sb.verifyHeader(chain, header, nil)
+	var parent []*types.Header
+	if header.Number.Cmp(common.Big0) == 0 {
```
Comment (bobpkr) [inline consensus/istanbul/backend/engine.go:176]: How about ...
```suggestion
	if header.Number.Sign() == 0 {
```

Comment (kjeom) [reply consensus/istanbul/backend/engine.go:176]: Good idea. Thanks!

**Review by bobpkr — APPROVED** [review]

File: consensus/istanbul/backend/engine.go [hunk 2]
```diff
@@ -172,7 +172,14 @@ func (sb *backend) computeSignatureAddrs(header *types.Header) error {
...
-	return sb.verifyHeader(chain, header, nil)
+	var parent []*types.Header
+	if header.Number.Sign() == 0 {
+		// If current block is genesis, the parent is also genesis
+		parent = append(parent, chain.GetHeaderByNumber(0))
+	} else {
+		parent = append(parent, chain.GetHeader(header.ParentHash, header.Number.Uint64()-1))
+	}
+	return sb.verifyHeader(chain, header, parent)
 }
```
Comment (jeongkyun-oh) [inline consensus/istanbul/backend/engine.go:183]: May I ask why this change is needed?

Comment (kjeom) [reply consensus/istanbul/backend/engine.go]: When getting validators for the startNewRound of consensus in `consensus/istanbul/core/core.go`, it calls snapshot that verify genesis block (ref. https://github.com/klaytn/klaytn/blob/dev/consensus/istanbul/backend/engine.go#L642).
- If you are asking why we need the parent as genesis in case of block number 0
  - Genesis block has no parent with hash, 0x000..., so we can use the genesis block as parent. I think it would has no problem.
- If you are asking the `else` case
  - After KIP-71 hardfork, we need the parent's header for child block's baseFee to verify that It is calculated correctly from the parent.

**Review by hqjang-pepper — APPROVED** [review]

**Review by jeongkyun-oh — APPROVED** [review]

**Review by jimni1222 — APPROVED** [review]
LGTM ! Thank you =)

**Review by hqjang-pepper — APPROVED** [review]
